Actions/RunPipeline/RunPipeline.ps1

```diff
@@ -152,11 +152,23 @@ try {
         })
     }
 
-    $installApps = $settings.installApps
-    $installTestApps = $settings.installTestApps
+    $install = @{
+        "apps" = $settings.installApps + @($installAppsJson | ConvertFrom-Json)
+        "testApps" = $settings.installTestApps + @($installTestAppsJson | ConvertFrom-Json)
+    }
 
-    $installApps += $installAppsJson | ConvertFrom-Json
-    $installTestApps += $installTestAppsJson | ConvertFrom-Json
+    # Replace secret names in install.apps and install.testApps
+    foreach($list in @('apps','testApps')) {
+        $install."$list" = @($install."$list" | ForEach-Object {
+            $pattern = '.*(\$\{\{\s*([^}]+?)\s*\}\}).*'
+            if ($_ -match $pattern) {
+                $_.Replace($matches[1],[System.Text.Encoding]::UTF8.GetString([System.Convert]::FromBase64String($secrets."$($matches[2])")))
```

Comment:
Would it make sense to check if the URL is valid (and the file accessible) here and fail with a useful error message? Something like this
`Invoke-WebRequest -Uri $_ -Method Head`

If your URL isn't valid for some reason (File was moved, link expired etc.) we might want to fail the run before we get too far.

Reply:
That sounds like a good idea - will add that.